src/main/java/com/napier/sem/DAO.java

```diff
@@ -193,6 +193,47 @@ public DAO(Connection connection)
 
         return ExecuteCityStatement(statementString);
     }
+
+    /**
+     * Use cases 4.1-4.5
+     * Constructs an SQL query to fetch the top N populated cities in a specific area, and executes the query.
+     *
+     * @return An ordered list of cities in a defined area sorted by descending population
+     */
+    public ArrayList<City> topNCitiesIn(String area, String areaName, Integer n)
+    {
+        String whereClause;
+
+        switch (area)
+        {
+            case CONTINENT:
+                whereClause = "WHERE country.continent = '" + areaName + "'\n";
+                break;
+            case REGION:
+                whereClause = "WHERE country.region = '" + areaName + "'\n";
+                break;
+            case COUNTRY:
+                whereClause = "WHERE country.name = '" + areaName + "'\n";
+                break;
+            case DISTRICT:
+                whereClause = "WHERE city.district = '" + areaName + "'\n";
+                break;
+            default:
+                whereClause = "";
+                break;
+        }
+
+        // Define the SQL query as a string
+        String statementString = "SELECT ID, city.name, district, city.population, country.name AS country\n" +
```

Comment:
we don't need `ID` in the select, as we don't use it. i'm happy to leave it for this though, there's no harm in it.

Reply:
Agreed. Removed it in the latest commit.